kml/serdes_gpx.cpp

```diff
@@ -209,23 +209,32 @@ void GpxParser::ParseGarminColor(std::string const & v)
   }
 }
 
+geometry::PointWithAltitude GpxParser::BuildPoint()
+{
+  geometry::PointWithAltitude point;
+  m2::PointD pt = mercator::FromLatLon(m_lat, m_lon);
+  point.SetPoint(pt);
+  point.SetAltitude(m_altitude);
+  return point;
+}
+
 void GpxParser::Pop(std::string_view tag)
 {
   ASSERT_EQUAL(m_tags.back(), tag, ());
   
   if (tag == gpx::kTrkPt || tag == gpx::kRtePt)
   {
-    m2::PointD p = mercator::FromLatLon(m_lat, m_lon);
-    if (m_line.empty() || !AlmostEqualAbs(m_line.back().GetPoint(), p, kMwmPointAccuracy))
-      m_line.emplace_back(std::move(p));
+    geometry::PointWithAltitude p = BuildPoint();
+    if (m_line.empty() || !AlmostEqualAbs(m_line.back().GetPoint(), p.GetPoint(), kMwmPointAccuracy))
+      m_line.emplace_back(p);
   }
   else if (tag == gpx::kTrkSeg || tag == gpx::kRte)
   {
     m_geometry.m_lines.push_back(std::move(m_line));
   }
   else if (tag == gpx::kWpt)
   {
-    m_org = mercator::FromLatLon(m_lat, m_lon);
+    m_org = BuildPoint();
```

Comment:
```suggestion
    m_org.setPoint(mercator::FromLatLon(m_lat, m_lon));
    m_org.setAltitude(m_altitude);    
```

Reply:
I suspect that we should reset m_altitude = kInvalidAltitude here too.

Reply:
Added reset and test UNIT_TEST(Gpx_Altitude_Issues)